File: csharp/ValidationProject/Product.cs
namespace Validation
{
    public class Product
    {
        public string Name { get; }
        public string Type { get; set; }
        public ProductFamily Family { get; set; }
        public ProductRange Range { get; set; }
        public double Weight { get; set; }

        public Product(string name)
        {
            Name = name;
        }

        public override string ToString()
        {
            return "Product{" +
                   "name='" + Name + '\'' +
                   ", type='" + Type + '\'' +
                   ", weight=" + Weight +
                   ", family=" + Family +
                   ", range=" + Range +
                   '}';
        }
    }

    public enum ProductRange
    {
        QUEEN,
        BUDGET,
        PROFESSIONAL
    }

    public enum ProductFamily
    {
        EYES,
        LIPS,
        LASHES,
        SKIN
    }
}

File: cpp/Product.cpp
#include "Product.h"

#include <utility>

Product::Product(std::string name) : name(std::move(name)) {
}

std::string Product::to_string() {
    std::string result;
    result.append("Product{");
    result.append("name='");
    result.append(name);
    result.append("\'");
    result.append(", type='");
    result.append(type);
    result.append("\'");
    result.append(", weight=");
    result.append(std::to_string(weight));
    result.append(", family=");
    result.append(std::to_string(family));
    result.append(", range=");
    result.append(std::to_string(range));
    result.append("}");
    return result;
}
